src/main/java/net/wiseoldman/features/AutoUpdateSession.java

```diff
@@ -0,0 +1,123 @@
+package net.wiseoldman.features;
+
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.events.GameTick;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.eventbus.Subscribe;
+import net.wiseoldman.WomUtilsPlugin;
+import net.wiseoldman.web.WomClient;
+
+import javax.inject.Inject;
+import java.util.Date;
+import java.util.HashMap;
+
+@Slf4j
+public class AutoUpdateSession {
+    @Inject
+    public Client client;
+
+    @Inject
+    public WomUtilsPlugin womUtilsPlugin;
+
+    @Inject
+    public WomClient womClient;
+
+    @Inject
+    private ClientThread clientThread;
+
+    /**
+     * The number of hours which a session lasts for.
+     */
+    private static final int HOURS_BETWEEN_SESSION = 24;
+
+    /**
+     * A record of players that have been updated for the current session.
+     * Key: Player name
+     * Value: Date of when the player should next be updated.
+     */
+    private final HashMap<String, Date> playersTracked = new HashMap<>();
```

Comment:
Do we need to support multiple accounts per session? With Jagex accounts becoming a thing, people won't be able to log in with more than one account per session anyway.

Reply:
I did this - as currently there is the possibility for multiple accounts to be on the same client.

All new accounts from November 2023 require it but suppose accounts made before then won't actually require it. I can't see any evidence that it will be "forced" at any specific date.

> As of right now, we do not know when that will be, but the time will come :). [Source](https://secure.runescape.com/m=forum/forums?408,409,406,66290645)

Happy to change it to only handle one account, but may cause issues for those who have not yet upgraded.